### PR TITLE
WIP: New SliderInput widget

### PR DESCRIPTION
**Pull request by xavArtley**
I open this PR mostly for discussion and improvements
The aim is to fix https://github.com/holoviz/panel/issues/1128 and https://github.com/holoviz/panel/issues/33

ezgif com-video-to-gif (3)

### CODE REVIEW

File: panel/models/widgets.py [hunk 1]
```diff
@@ -127,3 +127,22 @@ class FileDownload(InputWidget):
...
+
+
+class SliderInput(Slider):
+    """
+    Slider associate with an input
+    """
+    tooltips = Override(default=False)
+
+    hard_start = Float(default=None, help="""
+    The minimum allowable value accept by the input.
```
Comment (jbednar) [inline panel/models/widgets.py:139]:
```suggestion
    The minimum allowable value accepted as input, if any.
```

File: panel/models/widgets.py [hunk 2]
```diff
@@ -127,3 +127,22 @@ class FileDownload(InputWidget):
...
+    """)
+
+    hard_end = Float(default=None, help="""
+    The maximum allowable value accept by the input.
```
Comment (jbednar) [inline panel/models/widgets.py:143]:
```suggestion
    The maximum allowable value accepted as input, if any.
```

File: panel/models/widgets.py [hunk 3]
```diff
@@ -127,3 +127,22 @@ class FileDownload(InputWidget):
...
+    """)
+
+    input_size = String(default=None, help="""
+    Size of the input in px or %
```
Comment (jbednar) [inline panel/models/widgets.py:147]:
```suggestion
    Size of the widget in px or %
```
(Not sure about the name; seems like this is widget_size?  By input I'd expect to refer to the numeric value being put in, not the graphical widget.)

File: panel/widgets/slider.py [hunk 4]
```diff
@@ -113,6 +113,24 @@ class FloatSlider(ContinuousSlider):
 
     step = param.Number(default=0.1)
 
+class SliderInput(FloatSlider):
+
+    show_value = param.Boolean(default=False, doc="""
+        Whether to show the widget value.""")
```
Comment (jbednar) [inline panel/widgets/slider.py:119]:
```suggestion
        Whether to show the current value numerically alongside the widget.""")
```

File: panel/models/widgets.py (hunk 3 above)
Comment (xavArtley) [reply panel/models/widgets.py:147]: Actualy it's really just the size of the numeric input
the size of the overall widget is define by width

**Comment by xavArtley** [comment]
one of my concerns is the number of possibilities is the number of possibilities for the widget layout:
For example a small subset of possibilities

image

**Comment by jbednar** [comment]
Good point!  Of those, the ones I'd personally find useful are:

image
image
image

File: panel/models/widgets.py (hunk 3 above)
Comment (jbednar) [reply panel/models/widgets.py:147]: Ah! I consider both the slider and the text box to be ways of providing input, and so referring to one part of it as input confused me. :-) 

Maybe `slider_size` (if needed) and `textbox_size`?

**Comment by jbednar** [comment]
Also, I'd prefer the number to be "bare", as if it is merely reporting the value, unless and until someone clicks on it, as ipywidgets does:

image


image
 
Apps and dashboards tend to have a lot of sliders, only one of which is ever being messed with at a given time, and even then only rarely is it used numerically, so the numeric value is far more often used as a readout than as a widget.  So it would be great if the widgetness of it could fade into the background, and only be visible when someone actually tries to mess with it.

**Comment by xavArtley** [comment]
> Also, I'd prefer the number to be "bare", as if it is merely reporting the value, unless and until someone clicks on it, as ipywidgets does:
> 
> image
> 
> image
> 
> Apps and dashboards tend to have a lot of sliders, only one of which is ever being messed with at a given time, and even then only rarely is it used numerically as an input, so the numeric value is far more often used as a readout than as a widget. So it would be great if the widgetness of it could fade into the background, and only be visible when someone actually tries to mess with it.

ezgif com-video-to-gif (5)

**Comment by jbednar** [comment]
Beautiful!  You are a magical wish-granting genie! :-)

**Comment by Jhsmit** [comment]
Would it be a good idea to extend this SliderInput widget to have two fields on either side and a range slider for fine control of ranges?

**Comment by ahuang11** [comment]
Might be impossible in panel, but it'd be nice if the editable value box is directly above the slider:
image

https://github.com/bokeh/bokeh/issues/9943#issuecomment-718968158

**Pull request closed without merging**